src/main/java/org/entando/plugins/pda/serializer/JsonSchemaFormSerializer.java

```diff
@@ -155,25 +178,113 @@ private void writeFieldType(JsonGenerator jsonGenerator, String fieldType) throw
+            writeFieldInputList(jsonGenerator);
+        }
+    }
+
+    private void writeFieldNumberProperties(JsonGenerator jsonGenerator, FormFieldNumber field) throws IOException {
+        if (field.getType() == FormFieldType.INTEGER) {
+            if (field.getMinValue() != null) {
+                jsonGenerator.writeNumberField(TYPE_NUMBER_MIN, field.getMinValue().intValue());
+            }
+
+            if (field.getMaxValue() != null) {
+                jsonGenerator.writeNumberField(TYPE_NUMBER_MAX, field.getMaxValue().intValue());
             }
 
-            if (intField.getMaxValue() != null) {
-                jsonGenerator.writeNumberField(TYPE_INTEGER_MAX, intField.getMaxValue());
+            if (field.getMultipleOf() != null) {
+                jsonGenerator.writeNumberField(TYPE_NUMBER_MULTIPLE, field.getMultipleOf().intValue());
             }
-        } else if (field.getType().equals(FormFieldType.STRING)) {
-            FormFieldText strField = (FormFieldText) field;
-            if (strField.getMinLength() != null) {
-                jsonGenerator.writeNumberField(TYPE_STRING_MIN, strField.getMinLength());
+        } else {
+            if (field.getMinValue() != null) {
+                jsonGenerator.writeNumberField(TYPE_NUMBER_MIN, field.getMinValue());
             }
 
-            if (strField.getMaxLength() != null) {
-                jsonGenerator.writeNumberField(TYPE_STRING_MAX, strField.getMaxLength());
+            if (field.getMaxValue() != null) {
+                jsonGenerator.writeNumberField(TYPE_NUMBER_MAX, field.getMaxValue());
             }
+
+            if (field.getMultipleOf() != null) {
+                jsonGenerator.writeNumberField(TYPE_NUMBER_MULTIPLE, field.getMultipleOf());
+            }
+        }
+    }
+
+    private void writeFieldStringMinMax(JsonGenerator jsonGenerator, FormFieldText field) throws IOException {
+        if (field.getMinLength() != null) {
+            jsonGenerator.writeNumberField(TYPE_STRING_MIN, field.getMinLength());
+        }
+
+        if (field.getMaxLength() != null) {
+            jsonGenerator.writeNumberField(TYPE_STRING_MAX, field.getMaxLength());
         }
     }
+
+    private void writeFieldDateFormat(JsonGenerator jsonGenerator, FormFieldDate field) throws IOException {
+        if (field.isWithTime()) {
+            jsonGenerator.writeStringField(TYPE_DATE_FORMAT, DATE_TIME_FORMAT);
+        } else {
+            jsonGenerator.writeStringField(TYPE_DATE_FORMAT, DATE_FORMAT);
+        }
+    }
+
+    private void writeFieldSelector(JsonGenerator jsonGenerator, FormFieldSelector field) throws IOException {
+        if (StringUtils.isNotEmpty(field.getDefaultValue())) {
+            jsonGenerator.writeStringField(DEFAULT, field.getDefaultValue());
+        }
+
+        jsonGenerator.writeFieldName(ONE_OF);
+        jsonGenerator.writeStartArray();
+        for (FormFieldSelector.Option option : field.getOptions()) {
```

Comment:
This for loop is a duplication for the one in the method below, please resolve the duplication.